### PR TITLE
Have bot only report failures on maintained branches

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

Currently the QacoBot for reporting travis failures on commits is
reporting failures on temporary PR branches created on terra. There are
use cases for creating a quick branch to either test CI with secrets
(like to manually retrigger a failed wheel build) or a quick web ui
edit. However, failures on these branches should not have issues opened
against them because they are not maintained branches. This commit
updates the report_ci_failure.py script to only report a failure when
either master or a stable branch encounters a CI failure.

### Details and comments

### CODE REVIEW

File: tools/report_ci_failure.py [hunk 1]
```diff
@@ -46,6 +49,8 @@ def report(self, branch, commit, infourl=None, job_name=None):
                 build logs.
             job_name (str): name of the failed ci job.
         """
+        if branch != 'master' or not self._stable_regex.search(branch):
```
Comment (mtreinish) [inline tools/report_ci_failure.py:52]: This should be `and` not `or`
```suggestion
        if branch != 'master' and not self._stable_regex.search(branch):
```

**Comment by 1ucian0** [comment]
`on hold` for after the 0.14 release

File: tools/report_ci_failure.py [hunk 2]
```diff
@@ -46,6 +49,8 @@ def report(self, branch, commit, infourl=None, job_name=None):
                 build logs.
             job_name (str): name of the failed ci job.
         """
+        if branch != 'master' and not self._stable_regex.search(branch):
```
Comment (1ucian0) [inline tools/report_ci_failure.py:52]: why not just `branch.startswith('stable/')?

I hate regular expressions so I avoid them every time I can :)

File: tools/report_ci_failure.py [hunk 3]
```diff
@@ -60,7 +65,7 @@ def _key_label(self, branch_name, job_name):
             return 'benchmarks failing'
         elif branch_name == 'master':
             return 'master failing'
-        elif branch_name == 'stable':
+        elif 'stable' in branch_name:
```
Comment (1ucian0) [inline tools/report_ci_failure.py:68]: same here, for consistency.

Comment (1ucian0) [reply tools/report_ci_failure.py:68]: Added in 7ade64c29

File: tools/report_ci_failure.py (hunk 2 above)
Comment (1ucian0) [reply tools/report_ci_failure.py]: re removed in 160e13fce

**Pull request merged**